### PR TITLE
ceph.in: adjust usage width according to user's tty

### PR DESCRIPTION
**Pull request by tchaikov**
fixed a pep8 warning also

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: src/ceph.in [hunk 1]
```diff
@@ -359,36 +359,56 @@ def wrap(s, width, indent):
 
     raise StopIteration
 
+
+def terminal_width():
+    if not sys.stdout.isatty():
+        return 80
+    try:
+        import shutil
+        sz = shutil.get_terminal_size((80, 20))
```
Comment (dmick) [inline src/ceph.in]: Is this Python 3.x?

File: src/ceph.in [hunk 2]
```diff
@@ -359,36 +359,56 @@ def wrap(s, width, indent):
...
+        return sz.columns
+    except AttributeError:
+        import fcntl
+        import struct
+        import termios
+        arg = struct.pack('HHHH', 0, 0, 0, 0)
+        ret = fcntl.ioctl(sys.stdout.fileno(), termios.TIOCGWINSZ, arg)
```
Comment (dmick) [inline src/ceph.in]: We should probably figure out a way to reuse ceph_daemon.Termsize here (maybe moving it out of ceph_daemon)

File: src/ceph.in (hunk 1 above)
Comment (tchaikov) [inline src/ceph.in]: python 3.3 and up.

Comment (tchaikov) [inline src/ceph.in]: that's why i have a `except` clause below.

File: src/ceph.in [hunk 3]
```diff
@@ -359,36 +359,56 @@ def wrap(s, width, indent):
...
+        if flags & (FLAG_OBSOLETE | FLAG_DEPRECATED):
             continue
         concise = concise_sig(cmd['sig'])
         if partial and not concise.startswith(partial):
             continue
-        siglines = [l for l in wrap(concise, 40, 1)]
-        helplines = [l for l in wrap(cmd['help'], 39, 1)]
+        width = terminal_width() - 1  # 1 for the line between sig and help
+        sig_width = int(width / 2)
+        help_width = int(width / 2) + (width % 2)
```
Comment (dmick) [inline src/ceph.in]: Hm.   The intent of "-1" above was to handle the one-character indent of following lines; not sure that "+ (width % 2)" accomplishes that?

File: src/ceph.in [hunk 4]
```diff
@@ -359,36 +359,56 @@ def wrap(s, width, indent):
...
         # make lists the same length
         maxlen = max(len(siglines), len(helplines))
         siglines.extend([''] * (maxlen - len(siglines)))
         helplines.extend([''] * (maxlen - len(helplines)))
 
         # so we can zip them for output
-        for (s, h) in zip(siglines, helplines):
-            fullusage += '{0:40s} {1}\n'.format(s, h)
+        for s, h in zip(siglines, helplines):
+            fullusage += '{s:{w}s} {h}\n'.format(s=s, h=h, w=sig_width)
```
Comment (dmick) [inline src/ceph.in]: awesome, I did *not* know python's string formatting accepted variable width.  That's great to know.

File: src/ceph.in (hunk 1 above)
Comment (dmick) [inline src/ceph.in]: yep

File: src/ceph.in (hunk 3 above)
Comment (tchaikov) [inline src/ceph.in]: to make sure `sig_width + help_width == width`.

**Comment by tchaikov** [comment]
@dmick fixed and pushed.

File: src/pybind/ceph_daemon.py [hunk 5]
```diff
@@ -87,36 +87,37 @@ def do_sockio(path, cmd_bytes):
...
+        except IOError:
+            return self.DEFAULT_SIZE
+
     def update(self):
-        rows, cols = _gettermsize()
-        self.changed = self.changed or (
-            (self.rows != rows) or (self.cols != cols)
-        )
+        rows, cols = self._gettermsize()
+        if not self.changed:
```
Comment (dmick) [inline src/pybind/ceph_daemon.py]: You don't like my or?  FINE!  :)

File: src/pybind/ceph_daemon.py [hunk 6]
```diff
@@ -87,36 +87,37 @@ def do_sockio(path, cmd_bytes):
...
+            self.changed = (self.raw,self.col) != (rows, cols)
```
Comment (tchaikov) [inline src/pybind/ceph_daemon.py]: i confess: i just traded your `or`s with a bug.

Comment (tchaikov) [inline src/pybind/ceph_daemon.py]: https://github.com/ceph/ceph/pull/15253

Comment (dmick) [inline src/pybind/ceph_daemon.py]: heh. oops.

**Pull request merged**